Assets/MRTK/Providers/Oculus/XRSDK/MRTK.ThirdParty/MRTK-Quest/Scripts/Input/Controllers/OculusHand.cs

```diff
@@ -0,0 +1,591 @@
+
+                UpdateVelocity();
+            }
+
+            // UpdateTeleport();
+
+            for (int i = 0; i < Interactions?.Length; i++)
+            {
+                switch (Interactions[i].InputType)
+                {
+                    case DeviceInputType.SpatialPointer:
+                        Interactions[i].PoseData = currentPointerPose;
+                        if (Interactions[i].Changed)
+                        {
+                            CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction, currentPointerPose);
+                        }
+                        break;
+                    case DeviceInputType.SpatialGrip:
+                        Interactions[i].PoseData = currentGripPose;
+                        if (Interactions[i].Changed)
+                        {
+                            CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction, currentGripPose);
+                        }
+                        break;
+                    case DeviceInputType.Select:
+                        Interactions[i].BoolData = IsPinching || IsGrabbing;
+
+                        if (Interactions[i].Changed)
+                        {
+                            if (Interactions[i].BoolData)
+                            {
+                                CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
+                            }
+                            else
+                            {
+                                CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
+                            }
+                        }
+                        break;
+                    case DeviceInputType.TriggerPress:
+                        Interactions[i].BoolData = IsPinching || IsGrabbing;
+
+                        if (Interactions[i].Changed)
+                        {
+                            if (Interactions[i].BoolData)
+                            {
+                                CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
+                            }
+                            else
+                            {
+                                CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, Interactions[i].MixedRealityInputAction);
+                            }
+                        }
+                        break;
+                    case DeviceInputType.IndexFinger:
+                        UpdateIndexFingerData(Interactions[i]);
+                        break;
+                }
+            }
+        }
+
+        /*
```

Comment:
is this going to be uncommented? why is it commented out?

Reply:
It will be uncommented in a followup PR to add teleporter pointer features. I'll delete the commented out sections for now since it seems to be confusing and be careful when adding it back in.

Reply:
Updated with clear #if LATER markers